uwblueprint/planet-read
# Implement manage stories filter

## Notion ticket link



[Implement manage stories filter](https://www.notion.so/uwblueprintexecs/Implement-manage-stories-filter-8b37465fa9e04a3ea89cf136345e7398)



## Implementation description

- allow optional story title, start/end date upload filtering in get stories query
- add date filter to manage stories table



## Steps to test

1. Rebuild the containers 
2. Update the upload date of a few stories `docker exec -it planet-read_db_1 mysql -u root -proot -e "USE planet-read; UPDATE stories SET date_uploaded="2021-12-21 06:48:42" where id IN (4,5,6);"`; `docker exec -it planet-read_db_1 mysql -u root -proot -e "USE planet-read; UPDATE stories SET date_uploaded="2021-12-08 06:48:42" where id IN (1,2);"`; 
3. Login in as an admin
4. Navigate to the [manage stories tab](http://localhost:3000/#/?tab=1)
5. Play around with the date filter: if just one date is selected, stories with upload date >= than it should be returned; if a date range is selected, stories with an upload date within that selected time frame should be returned
6. Verify that filters on other tables still work as expected

![manage story date](https://user-images.githubusercontent.com/32009013/147863205-78bf73f8-86bc-4902-b47f-c7de24d00273.gif)




## What should reviewers focus on?

- Is there a better way to style the date picker?

## Checklist
